(Luis Mtz y Laura) Modificamos la funcion log y agregamos un test de la misma.

LauraMA opened the pull request:
Creo  que hay un error si al restringir el dominio se obtiene un None, ya que de cualquier modo se pide que se calcule el log de restringido por eso agregue un try. Creo que no sé qué es lo que debe de regresar la función en este caso pero me pareció que debía ser un "nan".

El test considera intervalos positivos (aunque aun requiere un chequeo ) y funciona, pero no sé si debe considerar en el test intervalos arbitrarios (negativos) ya que eso se controla  con la funcion log y su restringir dominio. 

Lo que falta checar es que sea compatible con la función exp ya que al hacer muchos nosetests aún saca error. Por cierto también el test de la exp tiene este problema.

davidphysdavalos committed:
Se *aleatorizaron* varios test definiendo una funcion auxiliar llamada TwoReals

LauraMA committed:
Merge branch 'master' of https://github.com/computo-fc/metodos_rigurosos

LauraMA committed:
estoy actualizando desde el upstream

LauraMA committed:
actualizando mi master local desde el upstream

LauraMA committed:
actualizacin menor

LauraMA committed:
(Luis Mtz y Laura) Modificamos el test_log. y la función log.

Sin embargo, falta checar que la función log  sea consistente con la
función exp, ya que esto es lo que falla en es test.
Además en la función log creo que habia un error cuando se quiere
calcular el log de un intervalo None, por ello se agregó que en este
caso regresara NaN (not aritmetic number).

LauraMA committed:
nada, creo que ya esta comentado.

lbenet commented:
Alguien cambió la función `TwoReals` para que acepte argumentos que son los límites. Creo que es mejor que usen esto en `test_intervalo.py`; deja el código más limpio. Espero su nueva implementación (ojo; han habido cambios que están en la nueva versión del repositorio).

No me quedó claro lo que no les queda claro del resultado de `log()`. Yo diría que si hay valores negativos restrinja el dominio a la intersección con [0,inf](y que avise de esto solamente si hay una restricción del dominio) y en este caso, `log(0)=-inf`; eso da consistencia con `exp(-inf)=0`. Es este el problema?

Otra cosa: Por favor NO incluyan en sus commits los archivos .pyc

LauraMA committed:
actualizacion de intervalo.py desde el master

LauraMA committed:
Modificacion-corrección  al test del logaritmo.

Analogamente al test de la división para checar la compatibilidad con la
fucion exp, solo pido que los resultados obtenidos sean muy cercanos.
Eliminé los docs  *.pyc

LauraMA committed:
Agregué un espacio sin importancia, pero este doc es igual que el anterior.

lbenet commented:
Corregí los tests de exp() y log() en `test_intervalo.py`. Ahora pasan bien. El error era llamarlos usando np.exp(a) o np.log(a); lo correcto es usar a.exp() y a.log() para ejecutar las funciones correspondientes de intervalos.
